runtime/buffer.h

```diff
@@ -252,9 +252,19 @@ class buffer : public raw_buffer {
   ~buffer() { free(); }
 
   buffer(const buffer&) = delete;
-  buffer(buffer&& m) = delete;
+  buffer(buffer&& m) { *this = std::move(m); }
   void operator=(const buffer&) = delete;
-  void operator=(buffer&& m) = delete;
+
+  void operator=(buffer&& m) {
+    memcpy((raw_buffer*)this, (const raw_buffer*)&m, sizeof(raw_buffer));
```

Comment:
C++ style casts (`static_cast<...>`), actually relevant in this case if we had multiple inheritance here :)

Reply:
It is verbose for sure! But I do like to be able to distinguish between `static_cast` vs. `reinterpret_cast`